### PR TITLE
[SimplifyCFG] Allow merging invoke's with different attrs

### PR DESCRIPTION
**Pull request by goldsteinn**
Same logic as other callsites, if the attributes are intersectable, we merge.

### CODE REVIEW

File: llvm/test/Transforms/SimplifyCFG/X86/merge-compatible-invokes-of-landingpad.ll [hunk 1]
```diff
@@ -1,4 +1,4 @@
-; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --check-attributes --check-globals
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --function-signature --check-attributes --check-globals
```
Comment (nikic) [inline llvm/test/Transforms/SimplifyCFG/X86/merge-compatible-invokes-of-landingpad.ll:1]: I'd suggest regenerating with `--version 5` instead.

**Review by nikic — APPROVED** [review]
LGTM

File: llvm/test/Transforms/SimplifyCFG/X86/merge-compatible-invokes-of-landingpad.ll (hunk 1 above)
Comment (goldsteinn) [reply llvm/test/Transforms/SimplifyCFG/X86/merge-compatible-invokes-of-landingpad.ll:1]: Sure, will regen before pushing.

**Review by dtcxzyw — APPROVED** [review]
LG

**Pull request closed without merging**